src/quorum/datadriven_test.rs

```diff
@@ -123,7 +123,76 @@ fn test_quorum(data: &TestData) -> String {
                 }
                 buf.push_str(&format!("{}\n", idx.0));
             } else {
-                // TODO(accelsao): port majority commit
+                use_group_commit = false;
+
+                let idx = c.committed_index(use_group_commit, &l);
+                buf.push_str(&c.describe(&l));
+
+                // Joining a majority with the empty majority should give same result.
+                let a_idx =
+                    JointConfig::new_joint_from_majorities(c.clone(), MajorityConfig::default())
+                        .committed_index(use_group_commit, &l);
+                if a_idx != idx {
+                    buf.push_str(&format!("{} <-- via zero-joint quorum\n", a_idx.0));
+                }
+
+                // Joining a majority with itself should give same result.
+                let a_idx = JointConfig::new_joint_from_majorities(c.clone(), c.clone())
+                    .committed_index(use_group_commit, &l);
+                if a_idx != idx {
+                    buf.push_str(&format!("{} <-- via self-joint quorum\n", a_idx.0));
+                }
+
+                // test overlaying
+                for id in c.raw_slice() {
+                    if let Some(iidx) = l.acked_index(id) {
+                        if idx.0 > iidx.index {
+                            // try index - 1
+                            l.insert(
```

Comment:
Why? It doesn't match the original implement.

Reply:
L154 is a typo, the original implement create a new `map` with `index - 1`, I think we can just insert a new Index instead of creating a new one.